src/commander.js

```diff
@@ -0,0 +1,348 @@
+                //            vellum.data.core.form.removeMugsFromForm([mug]);
+                //            vellum.refreshCurrentMug();
+                //            return mug;
+                //        }
+                //    }
+                //},
+                {
+                    // select question
+                    args: [questionRef],
+                    run: function (args) {
+                        var mug = args[0];
+                        if (mug) {
+                            vellum.setCurrentMug(mug);
+                            vellum.scrollTreeTo(mug.ufid);
+                            vellum.focusFirstInput();
+                            return mug;
+                        }
+                    }
+                },
+            ];
+
+        _.each(commandConfigs, function (commandConfig) {
+            var seen = [],
+                args = [],
+                index = forms.length;
+            _.each(commandConfig.args, function (arg) {
+                if (arg.names) {
+                    var parts = seen.concat(["(.*)$"]);
+                    forms.splice(index, 0, {
+                        regexp: new RegExp("^" + parts.join("\\s+"), "i"),
+                        items: _.map(arg.names, itemizer("")),
+                    });
+                }
+                seen.push("(?:" + arg.regexp + ")");
+                if (args.length < 1) {
+                    args.push("^(" + arg.regexp + ")");
+                } else {
+                    args.push("(?:\\s+(" + arg.regexp + ")");
+                }
+            });
+            args.push.apply(args, _.map(args, function () { return ")?"; }));
+            args[args.length - 1] = "$";
+            var tokenizer = new RegExp(args.join(""), "i");
+            tokenizer.config = commandConfig;
+            tokenizers.push(tokenizer);
+        });
+
+        return {
+            tokenize: tokenize,
+            dispatch: dispatch,
+            atwhoConfig: {
+                at: "",
+                data: [],
+                limit: 50,
+                maxLen: Infinity,
+                insertTpl: "${full}",
+                suffix: " ",
+                searchKey: $.fn.atwho["default"].searchKey,
+                tabSelectsMatch: true,
+                callbacks: {
+                    matcher: function (flag, subtext) { return subtext; },
+                    filter: filter,
+                    sorter: function(query, items) { return items; },
+                },
+            },
+        };
+    }
+
+    function onCommand(cmd) {
+        if (cmd.input.atwho('isSelecting')) {
+            // HACK delay until value has been inserted
+            setTimeout(function () { onCommand(cmd); }, 0);
+        } else {
+            var text = cmd.input.val(),
+                ok = fn.doCommand(text, cmd.vellum);
+            if (ok) {
+                hideCommander(cmd);
+            } else {
+                cmd.input.addClass("alert-danger");
```

Comment:
I'd like to do that in the future if I can think of a good place to show it. I put it off for now since it's a non-trivial feature involving more UI elements and I didn't want to get too distracted by that right now.